rust-lib/src/lib.rs

```diff
@@ -346,3 +355,51 @@ pub extern "C" fn transfer(
     )
     .into()
 }
+
+fn _asset_allocations(
+    runtime: &COpaqueStruct,
+    contract_id: *mut c_char,
+) -> Result<(), RequestError> {
+    let runtime = Runtime::from_opaque(runtime)?;
+
+    let c_contract_id = unsafe { CStr::from_ptr(contract_id) };
+    let contract_id = ContractId::from_str(c_contract_id.to_str()?)?;
```

Comment:
While `from_str` probably uses Bech32 encoding internally, previously there was an alternative form of `ContractId` representation as a hex string. So I propose to do more explicit conversion using `from_bech32_str` method (which also provides more clear error reporting) https://github.com/LNP-BP/rust-lnpbp/blob/master/src/rgb/bech32.rs#L219

Reply:
:+1: done